### PR TITLE
EES-5003 - Minor fixes

### PR DESCRIPTION
**Pull request by sambiramairelogic**
Quick intermediary PR which fixes a few niggles ahead of the next deploy to master -

- The `PUBLIC_URL_WITH_WWW` environment variable was incorrectly named, and has been corrected (this will require renaming the similarly incorrectly named pipeline variable).
- To combat the issue of the robots/sitemap files being constructed at build time, which only occurs once and does not have access to environment variables configured for each environment, we introduced `.env.production` so that the build would have access to the correct `PUBLIC_URL` both when running locally and on the pipeline. This however has consequences for the redirect tests, because the redirects are now going to the production url when testing locally. There's a few ways we could resolve this and keep `.env.production`, but this is all just getting too messy and hacky at this stage. I think the simplest thing is to revert `.env.production` and introduce a `PROD_PUBLIC_URL` value to the normal `.env` which only the sitemap config file uses. 
- The SEO metadata playwright tests have been flaky when running locally because the sitemap file produces xml which is confusing playwright's html locators. It finds an xml `<loc>` within an html `span`, so `getByText('foo')` finds two elements, as if both the loc and the span have the same innertext. I spent 15 minutes trying to resolve this but stopped after that since we've decided not to support Playwright tests. For now I've skipped the suite.

### CODE REVIEW

**Comment by sambiramairelogic** [comment]
FYI this is the playwright issue, seems to be using `#webkit-xml-viewer-source-xml` to parse the xml 
image

File: tests/robot-tests/.env.example
Comment (ntsim) [inline tests/robot-tests/.env.example:2]: We added a `user navigates to public frontend with www` keyword which means we don't need to add a new environment variable.

Can use this keyword for the UI test changes in this PR.

File: src/explore-education-statistics-frontend/.env
Comment (ntsim) [inline src/explore-education-statistics-frontend/.env:8]: Could get rid of the trailing slash given? We don't need to change this in all the environments or infrastructure so this should be pretty easy to do.

Will need adjustments in the `next-sitemap.config.js`

File: src/explore-education-statistics-frontend/next-sitemap.config.js
Comment (ntsim) [inline src/explore-education-statistics-frontend/next-sitemap.config.js:19]: Don't we need to make additional changes in the `server-sitemap.xml` as well? All the locations are using `PUBLIC_URL` instead of `PROD_PUBLIC_URL`

**Review by ntsim — CHANGES_REQUESTED** [review]

File: tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts
Comment (ntsim) [inline tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts:9]: Wouldn't it be better to skip just the offending test cases?

File: tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts
Comment (ntsim) [inline tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts:7]: Could use an XPath selector if there's an issue finding the correct element?

Also, it's weird that it's picking up a `span` because the sitemap doesn't actually contain this element!

File: tests/robot-tests/.env.example
Comment (sambiramairelogic) [reply tests/robot-tests/.env.example:2]: D'oh, yes of course we did. Will do

File: src/explore-education-statistics-frontend/next-sitemap.config.js
Comment (sambiramairelogic) [reply src/explore-education-statistics-frontend/next-sitemap.config.js:19]: Hah, yeah happily. I went with the slash to be consistent with `PUBLIC_URL` but yes would much prefer without

File: tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts
Comment (sambiramairelogic) [reply tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts:9]: Well, maybe. Ideally! 

Only one of them fails for me, but looking at the assertions in other tests I can't actually explain why the others don't fail too. They use `getByText...toHaveCount(1)` too. 
I'm just a bit concerned that this work has been causing an awful lot of pipeline failures, and these tests aren't really adding a supreme amount of value right now. Also, because the purpose of this PR is to iron out silly issues before a merge to master I'm trying to be as safe as possible. 

If you disagree then I'll happily just skip the one failing test - but wanted to share my reasoning

File: tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts
Comment (sambiramairelogic) [reply tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts:7]: Yep was considering that, alternatively could actually make an API GET request to the `/sitemap.xml` route with content type set to xml, then parse the response and assert on it all properly that way. Just cautious not to spend too long on Playwright without approval to do so 

And, yeah, really no idea why it's getting the span!

File: src/explore-education-statistics-frontend/.env
Comment (sambiramairelogic) [reply src/explore-education-statistics-frontend/.env:8]: Hah, yeah happily. I went with the slash to be consistent with PUBLIC_URL but yes would much prefer without

File: tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts
Comment (ntsim) [reply tests/playwright-tests/tests/public/seoMetadataFiles.spec.ts:9]: Okay fair enough, happy to leave it then!

**Review by ntsim — APPROVED** [review]

**Pull request merged**
